plugins/transformer/basic/client/recordupdater/record_updater.go

```diff
@@ -186,3 +258,49 @@ func (*RecordUpdater) obfuscateColumn(column arrow.Array) arrow.Array {
 	}
 	return bld.NewStringArray()
 }
+
+func (*RecordUpdater) obfuscateJSONColumn(column arrow.Array, jc jsonColumn) arrow.Array {
+	bld := types.NewJSONBuilder(array.NewExtensionBuilder(memory.DefaultAllocator, types.NewJSONType()))
+	for i := 0; i < column.Len(); i++ {
+		if !column.IsValid(i) {
+			bld.AppendNull()
+			continue
+		}
+
+		val := gjson.Get(column.ValueStr(i), jc.columnPath)
+		if val.Exists() && val.Type == gjson.String {
+			modified, err := sjson.Set(column.ValueStr(i), jc.columnPath, fmt.Sprintf("%x", sha256.Sum256([]byte(val.Str))))
+			if err == nil {
+				bld.AppendBytes([]byte(modified))
+			} else {
+				bld.AppendBytes([]byte(val.Str))
+			}
+			continue
+		}
+
+		bld.AppendBytes([]byte(fmt.Sprintf("%x", sha256.Sum256([]byte(column.ValueStr(i))))))
+	}
+	return bld.NewJSONArray()
+}
+
+func (r *RecordUpdater) splitJSONColumns(columnNames []string) (plainCols, jsonCols []string) {
```

Comment:
Let's say `columnNames` is `properties.impact, properties.resourceMetadata.resourceId, properties.extendedProperties.score`.

Won't this code return only one entry in `jsonCols` (`properties.extendedProperties.score` in the above case since it's the last item in the array)?

This results in not removing/obfuscating the other columns

Reply:
Yes, this was a major oversight. Should be fixed now. I don't have an Azure account handy but will test with similar AWS stuff.

Reply:
Added tests to cover this case.